src/snapshot/filesaver.js

```diff
@@ -0,0 +1,61 @@
+/*
+substantial portions of this code from FileSaver.js
+https://github.com/eligrey/FileSaver.js
+License: https://github.com/eligrey/FileSaver.js/blob/master/LICENSE.md
+*/
+
+/* FileSaver.js
+ * A saveAs() FileSaver implementation.
+ * 1.1.20160328
+ *
+ * By Eli Grey, http://eligrey.com
+ * License: MIT
+ *   See https://github.com/eligrey/FileSaver.js/blob/master/LICENSE.md
+ */
```

Comment:
Yes. Nicely done. You'll need to add this block:

```js
/**
 * Copyright 2012-2016, Plotly, Inc.
 * All rights reserved.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
```

at the top of the file to make `npm run test-syntax` happy.